main.cpp

```diff
@@ -388,12 +331,30 @@ struct IntType
     IntType& multiply( int rhs );
     IntType& divide( int rhs );
 
-    operator int() { return *value; }
+    IntType& pow(int rhs);
+    IntType& pow(const FloatType& rhs);
+    IntType& pow(const DoubleType& rhs);
+    IntType& pow(const IntType& rhs);
+
+    operator int() const { return *value; }
 
 private:
     int* value;
+    IntType& powInternal(int arg);
 };
 
+IntType& IntType::pow(int rhs)
+{
+    powInternal(rhs);
+    return *this;
+}
```

Comment:
what does `powInternal` return?
return the result of calling `powInternal`
```
return powInternal()
```
- [ ] revise